tests/Test/Client/_HiLoTest/HiLoTest.php

```diff
@@ -0,0 +1,213 @@
+                $session->saveChanges();
+
+                for ($i = 0; $i < 32; $i++) {
+                    $hiLoIdGenerator->generateDocumentId(new User());
+                }
+            } finally {
+                $session->close();
+            }
+
+            $session = $store->openSession();
+            try {
+                $hiloDoc = $session->load(HiloDoc::class, "Raven/Hilo/users");
+                $max = $hiloDoc->getMax();
+                $this->assertEquals(96, $max);
+
+                //we should be receiving a range of 64 now
+                $hiLoIdGenerator->generateDocumentId(new User());
+            } finally {
+                $session->close();
+            }
+
+            $session = $store->openSession();
+            try {
+                $hiloDoc = $session->load(HiloDoc::class, "Raven/Hilo/users");
+                $max = $hiloDoc->getMax();
+
+                // test fails here, we receive 128, not 160 from server as max - server doesn't respond with expected range
+                $this->assertEquals(160, $max);
+            } finally {
+                $session->close();
+            }
+        } finally {
+            $store->close();
+        }
+    }
+
+    public function testReturnUnusedRangeOnClose(): void
+    {
+        $store = $this->getDocumentStore();
+        try {
+
+            $newStore = new DocumentStore();
+            $newStore->setUrls($store->getUrls());
+            $newStore->setDatabase($store->getDatabase());
+
+            $newStore->initialize();
+
+            $session = $newStore->openSession();
+            try {
+                $hiloDoc = new HiloDoc();
+                $hiloDoc->setMax(32);
+                $session->store($hiloDoc, "Raven/Hilo/users");
+
+                $session->saveChanges();
+
+                $session->store(new User());
+                $session->store(new User());
+
+                $session->saveChanges();
+            } finally {
+                $session->close();
+            }
+
+            $newStore->close(); //on document store close, hilo-return should be called
+
+            $newStore = new DocumentStore();
+            $newStore->setUrls($store->getUrls());
+            $newStore->setDatabase($store->getDatabase());
+
+            $newStore->initialize();
+
+            $session = $newStore->openSession();
+            try {
+                $hiloDoc = $session->load(HiloDoc::class, "Raven/Hilo/users");
+                $max = $hiloDoc->getMax();
+                $this->assertEquals(34, $max);
+            } finally {
+                $session->close();
+            }
+
+            $newStore->close(); //on document store close, hilo-return should be called
+        } finally {
+            $store->close();
+        }
+    }
+
+    // we already are not implementing async call at the moment, so this test is not to be implemented at the moment
+    public function atestDoesNotGetAnotherRangeWhenDoingParallelRequests(): void
```

Comment:
can we use  $this->markTestSkipped here? So it is easier to discover such tests on CI?

Reply:
but I assume we remove this completely in future.

Reply:
Test marked as skipped